public/apps/configuration/panels/user-list.tsx

```diff
@@ -0,0 +1,199 @@
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+import {
+  EuiButton,
+  EuiContextMenuItem,
+  EuiContextMenuPanel,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiInMemoryTable,
+  EuiLink,
+  EuiPageBody,
+  EuiPageContent,
+  EuiPageContentHeader,
+  EuiPageContentHeaderSection,
+  EuiPageHeader,
+  EuiPopover,
+  EuiText,
+  EuiTitle,
+} from '@elastic/eui';
+import { difference, map, Dictionary } from 'lodash';
+import React, { useEffect, useState } from 'react';
+import { AppDependencies } from '../../types';
+import {
+  getUserList,
+  InternalUsersListing,
+  requestDeleteUsers,
+} from '../utils/internal-user-list-utils';
+
+function dictView() {
+  return (items: Dictionary<string>) => {
+    return (
+      <EuiFlexGroup direction="column" style={{ margin: '1px' }}>
+        {map(items, (v, k) => (
+          <EuiText key={k} size="xs">
+            {k}: {v}
+          </EuiText>
+        ))}
+      </EuiFlexGroup>
+    );
+  };
+}
+
+const columns = [
+  {
+    field: 'username',
+    name: 'Username',
+    sortable: true,
+  },
+  {
+    field: 'attributes',
+    name: 'Attributes',
+    render: dictView(),
+    truncateText: true,
+  },
+];
+
+export function UserList(props: AppDependencies) {
+  const [userData, setUserData] = useState<InternalUsersListing[]>([]);
+  const [errorFlag, setErrorFlag] = useState(false);
+  const [selection, setSelection] = useState<InternalUsersListing[]>([]);
+  const [isActionsPopoverOpen, setActionsPopoverOpen] = useState(false);
+  const [currentUsername, setCurrentUsername] = useState('');
+  // TODO: fetch current username
+  useEffect(() => {
+    const fetchData = async () => {
+      try {
+        setUserData(await getUserList(props.coreStart.http));
+      } catch (e) {
+        console.log(e);
+        setErrorFlag(true);
+      }
+    };
+
+    fetchData();
+  }, [props.coreStart.http]);
+
+  const handleDelete = async () => {
+    const usersToDelete: string[] = selection.map((r) => r.username);
+    try {
+      await requestDeleteUsers(props.coreStart.http, usersToDelete);
+      // Refresh from server (calling fetchData) does not work here, the server still return the roles
+      // that had been just deleted, probably because ES takes some time to sync to all nodes.
+      // So here remove the selected roles from local memory directly.
+      setUserData(difference(userData, selection));
+      setSelection([]);
+    } catch (e) {
+      console.log(e);
+    } finally {
+      setActionsPopoverOpen(false);
+    }
+  };
+
+  const actionsMenuItems = [
+    <EuiContextMenuItem
+      key="edit"
+      onClick={() => {}} // TODO: Redirect to edit page
+      disabled={selection.length !== 1}
+    >
+      Edit
+    </EuiContextMenuItem>,
+    // TODO: Redirect to duplicate page
+    <EuiContextMenuItem key="duplicate" onClick={() => {}} disabled={selection.length !== 1}>
+      Duplicate
```

Comment:
Looks like the mockup says "Export JSON" instead of "Duplicate"

Reply:
Huh, weird, there is a duplicate role page in the mockups:
![image](https://user-images.githubusercontent.com/63078162/83929985-48432580-a74a-11ea-98ec-0f696385a550.png)

Reply:
Not sure about the access point for this page. But from the dropdown mockup, this is how it looks like. Probably a follow up to the UX team.

<img width="433" alt="Screen Shot 2020-06-05 at 4 43 33 PM" src="https://user-images.githubusercontent.com/60111637/83930330-dcfa5300-a74b-11ea-8e5a-bd35a61d9a29.png">

Reply:
Sure, updated to Export JSON for current stage.